misk-aws/src/main/kotlin/misk/jobqueue/sqs/SqsJobConsumer.kt

```diff
@@ -45,91 +55,121 @@ internal class SqsJobConsumer @Inject internal constructor(
     }
 
     log.info {
-      "subscribing to queue ${queueName.value} with ${config.concurrent_receivers_per_queue} receivers"
+      "subscribing to queue ${queueName.value}"
     }
-
-    for (i in (0 until config.concurrent_receivers_per_queue)) {
-      taskQueue.scheduleWithBackoff(Duration.ZERO) {
-        // Don't call handlers until all services are ready, otherwise handlers will crash because
-        // the services they might need (databases, etc.) won't be ready.
-        serviceManagerProvider.get().awaitHealthy()
-        receiver.runOnce()
-      }
+    taskQueue.scheduleWithBackoff(Duration.ZERO) {
+      // Don't call handlers until all services are ready, otherwise handlers will crash because
+      // the services they might need (databases, etc.) won't be ready.
+      serviceManagerProvider.get().awaitHealthy()
+      receiver.runOnce()
     }
   }
 
   internal fun getReceiver(queueName: QueueName): QueueReceiver {
     return subscriptions[queueName]!!
   }
 
-  internal inner class QueueReceiver(
-    private val queueName: QueueName,
+  internal inner class QueueReceiver (
+    queueName: QueueName,
     private val handler: JobHandler
   ) {
     private val queue = queues[queueName]
 
     fun runOnce(): Status {
-      val (receiveDuration, messages) = queue.call { client ->
-        val receiveRequest = ReceiveMessageRequest()
-          .withAttributeNames("All")
-          .withMessageAttributeNames("All")
-          .withQueueUrl(queue.url)
-          .withMaxNumberOfMessages(config.message_batch_size)
-
-        timed { client.receiveMessage(receiveRequest).messages }
+      // Receive messages in parallel. Default to 1 if feature flag is not defined.
+      val numReceivers = max(featureFlags.getInt(RECEIVERS_PER_QUEUE, queue.name.value), 1)
+      val futures = (1..numReceivers)
+          .filter { num ->
+            val lease = leaseManager.requestLease("sqs-job-consumer-${queue.name.value}-$num")
+            config.clustered_consumers || lease.checkHeld()
```

Comment:
Looking at the description of `clustered_consumers`, I believe we'd want this `.filter` predicate to return `true` when the flag is either off or on-and-lease-held.

I'd rewrite to skip the lease check  when `clustered_consumers`  is set to `false`; otherwise return `true` if lease is held.

Reply:
yeah, good catch! I think I need to rewrite this to: 
```
!config.clustered_consumers || lease.checkHeld()
```

Reply:
Yup! That or returning early if `custered_consumers` is off (i.e. don't even request & check lease).